tests/Backend/Snowflake/CloneInputMappingTest.php

```diff
@@ -0,0 +1,312 @@
+    }
+
+    public function testClonePreserve()
+    {
+        $sourceBucketId = $this->getTestBucketId();
+        $sourceTableId = $this->createTableFromFile(
+            $this->_client,
+            $sourceBucketId,
+            self::IMPORT_FILE_PATH
+        );
+
+        $workspacesClient = new Workspaces($this->_client);
+        $workspace = $workspacesClient->createWorkspace([
+            'name' => 'cloning',
+        ]);
+
+        // first load
+        $workspacesClient->cloneIntoWorkspace($workspace['id'], [
+            'input' => [
+                [
+                    'source' => $sourceTableId,
+                    'destination' => 'languagesDetails',
+                ],
+            ],
+        ]);
+
+        // second load with preserve
+        $workspacesClient->cloneIntoWorkspace($workspace['id'], [
+            'input' => [
+                [
+                    'source' => $sourceTableId,
+                    'destination' => 'languages-2',
+                ],
+            ],
+            'preserve' => true,
+        ]);
+
+        $backend = WorkspaceBackendFactory::createWorkspaceBackend($workspace);
+        $backendTables = $backend->getTables();
+        $this->assertEquals(
+            [
+                'languages-2',
+                'languagesDetails',
+            ],
+            $backendTables
+        );
+    }
+
+    public function aliasSettingsProvider(): array
+    {
+        return [
+            'filtered alias' => [
+                [
+                    'aliasFilter' => [
+                        'column' => 'id',
+                        'values' => [26],
+                    ],
+                ],
+            ],
+            'selected columns' => [
+                [
+                    'aliasColumns' => [
+                        'id',
+                    ],
+                ],
+            ],
+        ];
+    }
+
+    private function ensureBucket(
+        Client $client,
+        string $bucketName,
+        string $stage = 'in'
+    ): string {
+    
+        if ($client->bucketExists($stage . '.' . 'c-' . $bucketName)) {
+            $client->dropBucket(
+                $stage . '.' . 'c-' . $bucketName,
+                [
+                    'force' => true,
+                ]
+            );
+        }
+        return $client->createBucket($bucketName, $stage);
+    }
+
+    /**
+     * @param array|string $primaryKey
+     */
+    private function createTableFromFile(
+        Client $client,
+        string $bucketId,
+        string $importFilePath,
+        $primaryKey = 'id'
+    ): string {
+    
+        return $client->createTable(
+            $bucketId,
+            'languagesDetails',
+            new CsvFile($importFilePath),
+            ['primaryKey' => $primaryKey]
+        );
+    }
+
+    /**
+     * @param $tableId
+     * @param $workspacesClient
+     * @param $workspace
+     */
+    private function runAndAssertWorkspaceClone($tableId): void
```

Comment:
Trochu se mi nepozdává to skládání testů, `ensureBucket` + `runAndAssertWorkspaceClone`, funkčně je to v pohodě, ale jako by si to říkalo o reorganizaci. Některý testy to používají, některý ne, už jen to spojení `runAndAssert` je podivný. Jak to ale fixnout nevím, od boku bych střelil vlastní testsuitu a dataprovidery?

Reply:
jo to ještě upravím ať je to standardněji oproti jiným testům

Reply:
Upravil jsem to přes dataProvider ale nejsem si jistej jestli je to lepší, teď je v tom testu zas podmínka:(